pkg/git/repository.go

```diff
@@ -212,30 +213,51 @@ func (r *repo) Persist(ctx context.Context, opts *PushOptions) (string, error) {
 		progress = r.progress
 	}
 
-	addPattern := "."
+	h, _, err := r.commit(opts)
+	if err != nil {
+		return "", err
+	}
+
+	return h.String(), r.PushContext(ctx, &gg.PushOptions{
+		Auth:     getAuth(r.auth),
+		Progress: progress,
+	})
+}
 
-	if opts.AddGlobPattern != "" {
-		addPattern = opts.AddGlobPattern
+func (r *repo) commit(opts *PushOptions) (plumbing.Hash, gogit.Worktree, error) {
+	var h plumbing.Hash
+
+	cfg, err := r.ConfigScoped(config.SystemScope)
+	if err != nil {
+		return h, nil, err
```

Comment:
whenever there is an error, return `nil, nil, err`, (also in the next lines)
if `plumbing.Hash` can not be nil (if it's a struct) - just change the return type to be `*plumbing.Hash`. I don't think you will need to make any changes in the calls.